### PR TITLE
Template location ignored if html if defined

### PR DESCRIPTION
**Pull request by RaananW**
The template location is ignored (mainly since the elimination of IDs in the viewer).
This fix provides two changes:
1) When html is an empty string, location will be used.
2) DOM Mapper will parse undefined and null correctly. so setting a parameter to "undefined" (as string) will set it to undefined (the type)

### CODE REVIEW

File: Viewer/src/configuration/mappers.ts [hunk 1]
```diff
@@ -52,6 +52,10 @@ class HTMLMapper implements IMapper {
                         val = true;
                     } else if (val === "false") {
                         val = false;
+                    } else if (val === "undefeind") {
```
Comment (sebavan) [inline Viewer/src/configuration/mappers.ts]: are you sure it should not be undefined ?

Comment (RaananW) [reply Viewer/src/configuration/mappers.ts]: yes I am. this is the mapper, which maps strings from the dom attributed to real objects

Comment (RaananW) [reply Viewer/src/configuration/mappers.ts]: so people will be able to set elements to be undefined (for example - templates.nav-bar.html="undefined")

Comment (sebavan) [reply Viewer/src/configuration/mappers.ts]: but undefeind and not undefined looks like a typo ?

**Pull request merged**